lib/gui/modules/drive-scanner.js

```diff
@@ -16,100 +16,144 @@
-  })
-
-  // Build human friendly "description"
-  .map((drives) => {
-    return _.map(drives, (drive) => {
-      drive.name = drive.device;
+/**
+ * @summary DriveScanner class
+ * @class
+ * @extends {EventEmitter}
+ * @example
+ * var scanner = new DriveScanner()
+ *   .on('error', (error) => { ... })
+ *   .on('drives', (drives) => { ... })
+ */
+class DriveScanner extends EventEmitter {
+
+  /**
+   * @summary DriveScanner constructor
+   * @param {Object} [options] - EventEmitter options
+   * @example
+   * var scanner = new DriveScanner()
+   *   .on('error', (error) => { ... })
+   *   .on('drives', (drives) => { ... })
+   */
+  constructor(options) {
+    super(options);
+    this.isRunning = false;
+    this.timer = null;
+    this.run = this.run.bind(this);
+  }
+
+  /**
+   * @summary Start the scanner
+   * @public
+   * @example
+   * scanner.start()
+   */
+  start() {
+    debug('start');
+    this.isRunning = true;
+    this.timer = setTimeout(this.run, DRIVE_SCANNER_FIRST_SCAN_DELAY_MS);
+  }
+
+  /**
+   * @summary Kick off a scan & schedule the next run
+   * @private
+   * @example
+   * scanner.run()
+   */
+  run() {
+
+    if (!this.isRunning) {
+      return;
+    }
 
-      if (os.platform() === 'win32' && !_.isEmpty(drive.mountpoints)) {
-        drive.name = _.join(_.map(drive.mountpoints, 'path'), ', ');
+    DriveScanner.scan((error, drives) => {
+      debug('scan', error || drives);
+      if (error) {
+        this.emit('error', error);
+      } else {
+        this.emit('drives', drives);
+      }
+      if (this.isRunning) {
+        this.timer = setTimeout(this.run, DRIVE_SCANNER_DELAY_MS);
       }
-
-      return drive;
     });
-  })
 
-  .map((drives) => {
-    if (settings.get('unsafeMode')) {
-      return drives;
-    }
+  }
 
-    return _.reject(drives, {
-      system: true
-    });
-  })
-  .pausable(new Rx.Subject());
+  /**
+   * @summary Stop scanning for drives
+   * @public
+   * @example
+   * scanner.stop()
+   */
+  stop() {
+    debug('stop');
+    this.isRunning = false;
+    clearTimeout(this.timer);
+  }
 
-/*
- * This service emits the following events:
- *
- * - `drives (Object[])`
- * - `error (Error)`
- *
- * For example:
- *
- * ```
- * driveScanner.on('drives', (drives) => {
- *   console.log(drives);
- * });
- *
- * driveScanner.on('error', (error) => {
- *   throw error;
- * });
- * ```
- */
-availableDrives.subscribe((drives) => {
-  emitter.emit('drives', drives);
-}, (error) => {
-  emitter.emit('error', error);
-});
+}
 
 /**
- * @summary Start scanning drives
- * @function
- * @public
- *
+ * @summary Post-process discovered drive objects
+ * @private
+ * @param {Array<Object>} drives - list of discovered drives
+ * @returns {Array<Object>}
  * @example
- * driveScanner.start();
+ * drives = DriveScanner.processDrives(drives);
  */
-emitter.start = () => {
-  availableDrives.resume();
+DriveScanner.processDrives = (drives) => {
```

Comment:
Yeah, I also had it in there first, but the linter complained about it being a class method without using `this`, so here it is now.